src/pages/login/hooks/useLoginForm.ts

```
@@ -2,13 +2,26 @@ import { useForm } from 'react-hook-form';
+    return typeof msg === 'string' ? msg : undefined;
+  }
+  return undefined;
+}
 
 export const useLoginForm = () => {
   const routing = usePageRouting();
-  const { login } = useAuth();
+  const { mutate } = useLoginMutation();
+  const location = useLocation();
+  const navigate = useNavigate();
+  const from = (location.state as any)?.from?.pathname ?? PATH.HOME;
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)